src/sql-parser/src/parser.rs

```diff
@@ -1707,9 +1707,9 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_drop(&mut self) -> Result<Statement, ParserError> {
-        let object_type = match self
-            .parse_one_of_keywords(&[DATABASE, SCHEMA, TABLE, VIEW, SOURCE, SINK, INDEX, TYPE])
-        {
+        let object_type = match self.parse_one_of_keywords(&[
+            DATABASE, SCHEMA, TABLE, VIEW, SOURCE, SINK, INDEX, TYPE, OBJECT,
```

Comment:
Seems odd to me to create a keyword for one case where we want to disallow the statement, and we didn't add an equivalent "CREATE" catch.

Reply:
I think we still need the keyword for the SHOW case. Regardless I don't think we actually need to handle `OBJECT` specially here; the error message for `DROP LAMP` is probably good enough for `DROP OBJECT`.